client/common/termutils.js

```diff
@@ -84,6 +109,39 @@ export function sample_match_termvaluesetting(row, filter) {
 				}
 			} else if (t.term.type == 'survival') {
 				// don't do anything?
+			} else if (t.term.type == 'geneVariant' && t.legendFilterType == 'geneVariant_hard') {
+				// handle a matrix legend hard filter
+				// values: [{ dt, origin, mclasslst:[key] }]
+				const f = t.values[0] //matrix geneVariant legend filter only has one item in tvs.values
+				thistermmatch =
+					samplevalue.find(s => {
+						for (const v of s.values) {
+							if (v.dt == f.dt && (v.origin ? v.origin == f.origin : true) && f.mclasslst.includes(v.class)) return true
+						}
+					}) && true
+			} else if (t.term.type == 'geneVariant' && t.legendFilterType == 'geneVariant_soft') {
```

Comment:
@congyu-lu same comment as above, no flag needed, processing should be determined by the filter data. There will be a difference with negation and join operators.

Reply:
will improve in another branch